### PR TITLE
Llvm17ray

### PR DESCRIPTION
**Pull request by nishant-sachdeva**

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -10,7 +10,7 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O2")
 set(LT_LLVM_INSTALL_DIR "" CACHE PATH "LLVM installation directory")
 list(APPEND CMAKE_PREFIX_PATH "${LT_LLVM_INSTALL_DIR}/lib/cmake/llvm/")
 
-find_package(LLVM 16.0.0 REQUIRED CONFIG)
+find_package(LLVM 17.0.0 REQUIRED CONFIG)
```
Comment (svkeerthy) [inline src/CMakeLists.txt:13]: Should it be 17.0.0? Can you please recheck

File: seed_embeddings/OpenKE/analogy.py [hunk 2]
```diff
@@ -57,9 +57,7 @@ def findTopk(dict1, k, values):
 
 
 def getAnalogyScore(fileName):
-    refFile = (
-        "/home/intern23002/iitH/ir2vec/IR2Vec/seed_embeddings/OpenKE/analogies.txt"
-    )
+    refFile = "/home/nishu/nishant/ir2vec/IR2Vec/seed_embeddings/OpenKE/analogies.txt"
```
Comment (svkeerthy) [inline seed_embeddings/OpenKE/analogy.py:60]: Can you change it to relative path?

File: seed_embeddings/OpenKE/generate_embedding_ray.py [hunk 3]
```diff
@@ -126,7 +126,7 @@ def findRep(src, dest, index_dir):
         metavar="DIRECTORY",
         help="Location of the directory entity2id.txt, train2id.txt and relation2id.txt",
         required=False,
-        default="/home/intern23002/iitH/ir2vec/IR2Vec/seed_embeddings/preprocessed/",
+        default="/home/nishu/nishant/ir2vec/IR2Vec/seed_embeddings/preprocessed/",
```
Comment (svkeerthy) [inline seed_embeddings/OpenKE/generate_embedding_ray.py:129]: Please change it to relative path.

**Review by svkeerthy — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -18,7 +18,8 @@ Please see [here](https://compilers.cse.iith.ac.in/projects/ir2vec/) for more de
 
 | LLVM Version | Branch |
 | ------------ | ------ |
-| LLVM 16.0.1 | [main](https://github.com/IITH-Compilers/IR2Vec) |
+| LLVM 17.0.2 | [main](https://github.com/IITH-Compilers/IR2Vec) |
```
Comment (svkeerthy) [inline README.md:21]: Should we move it to v17.0.6?

**Review by svkeerthy — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (nishant-sachdeva) [reply README.md:21]: The Docker image uses `17.0.2`, so I thought we should keep it that way

File: src/CMakeLists.txt (hunk 1 above)
Comment (nishant-sachdeva) [reply src/CMakeLists.txt:13]: As far as I understand, this ensures that a minimum of llvm17.0.0 should be used. Future versions of llvm17 also compile with this option. So ,  I think it should be 17.0.0

File: seed_embeddings/OpenKE/analogy.py (hunk 2 above)
Comment (nishant-sachdeva) [reply seed_embeddings/OpenKE/analogy.py]: Done

File: seed_embeddings/OpenKE/generate_embedding_ray.py (hunk 3 above)
Comment (nishant-sachdeva) [reply seed_embeddings/OpenKE/generate_embedding_ray.py]: Done

Comment (nishant-sachdeva) [reply seed_embeddings/OpenKE/generate_embedding_ray.py]: Done

File: README.md (hunk 4 above)
Comment (svkeerthy) [reply README.md:21]: Let us just check once with 17.0.6 please.

Comment (nishant-sachdeva) [reply README.md]: Done

**Review by svkeerthy — APPROVED** [review]

**Pull request merged**